include/openssl/byteorder.h

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2019-2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OPENSSL_BYTEORDER_H
+# define OPENSSL_BYTEORDER_H
+# pragma once
+
+# include <openssl/e_os2.h>
+
+/*
+ * "Modern" compilers do a decent job of optimising these functions to just a
+ * couple of instruction ([swap +] store, or load [+ swap]) when either no
+ * swapping is required, or a suitable swap instruction is available.
+ */
+
+static ossl_inline ossl_unused uint8_t *
+OSSL_CRYPTO_store_u16_le(uint8_t *out, uint16_t val)
```

Comment:
The `OSSL_CRYPTO` is a weird prefix for these. We usually use `OPENSSL_` prefix for _general use_ functions such as OPENSSL_malloc(), OPENSSL_strcasecmp(), etc. If this is to be used in both ML-KEM and ML-DSA branches, it should be separated out in an independent PR against the master branch, IMO.

Reply:
I've opened a PR against master with he `byteorder` code, let's sort this out there, and then adopt here.